tests/std/tests/P0768R1_spaceship_operator/test.cpp

```diff
@@ -187,7 +179,101 @@ void test_algorithm() {
     assert((test_algorithm2<Ty1, Ty2>()));
     assert((test_algorithm2<Ty2, Ty1>()));
 }
-#endif // defined(__cpp_impl_three_way_comparison) && defined(__cpp_lib_concepts)
+#endif // __cpp_lib_concepts
+
+// Guard against regression of GH-1050: 0 <=> partial_ordering::unordered returns invalid value
+static_assert(!(std::partial_ordering::less == 0));
+static_assert(std::partial_ordering::less != 0);
+static_assert(std::partial_ordering::less < 0);
+static_assert(!(std::partial_ordering::less > 0));
+static_assert(std::partial_ordering::less <= 0);
+static_assert(!(std::partial_ordering::less >= 0));
+
+static_assert(std::partial_ordering::equivalent == 0);
+static_assert(!(std::partial_ordering::equivalent != 0));
+static_assert(!(std::partial_ordering::equivalent < 0));
+static_assert(!(std::partial_ordering::equivalent > 0));
+static_assert(std::partial_ordering::equivalent <= 0);
+static_assert(std::partial_ordering::equivalent >= 0);
+
+static_assert(!(std::partial_ordering::greater == 0));
+static_assert(std::partial_ordering::greater != 0);
+static_assert(!(std::partial_ordering::greater < 0));
+static_assert(std::partial_ordering::greater > 0);
+static_assert(!(std::partial_ordering::greater <= 0));
+static_assert(std::partial_ordering::greater >= 0);
+
+static_assert(!(std::partial_ordering::unordered == 0));
+static_assert(std::partial_ordering::unordered != 0);

```

Comment:
This seems like it may be a mistake that WG21 introduced in P1185. I will investigate.

Reply:
> This seems like it may be a mistake that WG21 introduced in P1185. I will investigate.

It was P1614R2, not P1185R2, which removed the `!=` operators for the comparison category types. This was the behavior before that change as well. 🤷

Reply:
Doesn't this imitate the behavior of floating-point NANs, where they compare non-equal to everything else?

Reply:
This `static_assert` has held all the way back to P0515R3. It doesn't match my intuition, either.